Engine/DataFeeds/Enumerators/TradeBarBuilderEnumerator.cs

```diff
@@ -79,6 +79,13 @@ public void ProcessData(BaseData data)
                 var utcNow = _timeProvider.GetUtcNow();
                 var currentLocalTime = utcNow.ConvertFromUtc(_timeZone);
                 var barStartTime = currentLocalTime.RoundDown(_barSize);
+
+                if (barStartTime == currentLocalTime)
```

Comment:
Is this correct? I think the builder enumerators construct bars on the interval `[start, end)` which implies that a tick that falls on an interval boundary belongs to the following interval and not the preceding interval. I think this is confirmed by the usage of `<=` [here](https://github.com/QuantConnect/Lean/pull/3381/files#diff-966cb3974d100f934938a28c9b66e9eaR124) and [here](https://github.com/QuantConnect/Lean/pull/3381/files#diff-7005b62745f9c26d610567b6c5c5b62bR119). Also, if you look at the implementation of the consolidators, they too would appear to treat a data point on an interval boundary as belonging to the following bar and not the preceding bar.

Also, I may just be reading this wrong, and if that's the case, let's add some more comments here to better describe why we're doing this and why it's the right thing to do.

Maybe more importantly for me (and the code base in general), is to confirm and document the rules around aggregation for posterity (and perhaps in the source as well).

Reply:
@mchandschuh You're right, the aggregation interval is open on the right, so this change is incorrect, as @Martin-Molinero also pointed out. This needs to be taken into account when writing unit tests with ticks on the interval boundary -- like `AggregatesHourlyTicksIntoHourlyBars`.